packages/iov-bns/src/types.ts

```diff
@@ -282,65 +267,79 @@ export interface RegisterBlockchainTx extends BnsTx {
   readonly codecConfig: string;
 }
 
-export interface RegisterUsernameTx extends BnsTx {
-  readonly kind: TransactionKind.RegisterUsername;
+export interface RegisterUsernameTx extends UnsignedTransaction {
+  readonly domain: bnsDomain;
+  readonly kind: "register_username";
   readonly username: string;
   readonly addresses: ReadonlyArray<ChainAddressPair>;
 }
 
-export interface RemoveAddressFromUsernameTx extends BnsTx {
-  readonly kind: TransactionKind.RemoveAddressFromUsername;
+export interface RemoveAddressFromUsernameTx extends UnsignedTransaction {
+  readonly domain: bnsDomain;
+  readonly kind: "remove_address_from_username";
   /** the username to be updated, must exist on chain */
   readonly username: string;
   readonly payload: ChainAddressPair;
 }
 
+export type BnsTx =
+  | AddAddressToUsernameTx
+  | SendTx
+  | SetNameTx
+  | SwapOfferTx
+  | SwapCounterTx
+  | SwapClaimTx
+  | SwapTimeoutTx
+  | RegisterBlockchainTx
+  | RegisterUsernameTx
+  | RemoveAddressFromUsernameTx;
+
 export function isBnsTx(transaction: UnsignedTransaction): transaction is BnsTx {
-  return transaction.domain === bnsDomain;
+  return (transaction as BnsTx).domain === "bns";
```

Comment:
Do you need this `as BnsTx` here?
Unless you just changed it, `UnsignedTransaction` should have a `domain: string` member, which is good enough for a check

Reply:
Should indeed not be necessary